File: demo/functions/basic/divide-by/Handler.fs
namespace Calc

open Amazon.Lambda.Core

[<assembly:LambdaSerializer(typeof<Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer>)>]
do ()

[<CLIMutable>]
type Request = { _fal : string }

type KeyNotFound = System.Collections.Generic.KeyNotFoundException


module DivideBy =
    open System
    open System.IO
    open System.Text

    let run(req : Request) =
        printfn "Got event: %A" req

        if (req._fal = null)
            then raise (KeyNotFound("Event is missing \"_fal\" property"))

        let input = int req._fal
        let output = string(input / 2)

        { _fal = output }

File: demo/functions/basic/sleep/Handler.cs
using System;
using System.Threading;

[assembly:LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]

namespace Basic;

public class Sleep
{
   public void Run(Request req)
   {
       var seconds = req._fal is string && req._fal.Length > 0
           ? Convert.ToInt32(req._fal)
           : 5;

       Console.WriteLine("Seconds to sleep: " + seconds);

       Thread.Sleep(seconds * 1000);
       Console.WriteLine("Finished");
   }
}

public class Request
{
    public string _fal {get; set;}
}
